vispy/visuals/volume.py

```diff
@@ -51,37 +51,15 @@
 attribute vec3 a_position;
 attribute vec3 a_texcoord;
 uniform vec3 u_shape;
-uniform float u_cameraclip;
 
 varying vec3 v_texcoord;
-varying vec3 v_ray;
+varying vec3 v_position;
 varying vec4 v_clipplane;
 
 void main() {
     v_texcoord = a_texcoord;
-    gl_Position = $transform(vec4(a_position, 1.0));
-    
-    // Project local vertex coordinate to camera position. Then do a step
-    // backward (in cam coords) and project back. Voila, we get our ray vector.
-    // This vector does not interpolate nicely between vertices when 
-    // we have perspective view transform. Therefore we create a grid of 
-    // vertices rather than one quad per face.
-    vec4 pos_in_cam1 = $viewtransformf(vec4(a_position, 1.0));
-    vec4 pos_in_cam2 = pos_in_cam1 + vec4(0.0, 0.0, 1.0, 0.0); // step backward
-    vec4 position2 = $viewtransformi(vec4(pos_in_cam2.xyz, 1.0));
-    
-    // Calculate ray. In the fragment shader we do another normalization
-    // and scale for texture coords; interpolation does not maintain
-    // vector length
-    v_ray = normalize(a_position - position2.xyz);
-    
-    // Calculate a clip plane (in texture coordinates) for the camera
-    // position, in case that the camera is inside the volume.
-    pos_in_cam1.zw = vec2(u_cameraclip, 1.0);
-    vec3 cameraposinvol = $viewtransformi(pos_in_cam1).xyz;
-    cameraposinvol /= u_shape;  // express in texture coords
-    v_clipplane.xyz = v_ray;
-    v_clipplane.w = dot(v_clipplane.xyz, cameraposinvol);
+    v_position = a_position;
+    gl_Position = $transform(vec4(v_position, 1));
```

Comment:
Yes, we've had problems with some drivers because of this. Also, it will *always* cause compilation errors on WebGL.